smt/surrogate_models/krg_based.py

```diff
@@ -187,24 +234,27 @@ def _reduced_likelihood_function(self, theta):
         reduced_likelihood_function_value = -np.inf
         par = {}
         # Set up R
-        MACHINE_EPSILON = np.finfo(np.double).eps
-        nugget = 10.0 * MACHINE_EPSILON
-        if self.name == "MFK":
-            if self._lvl != self.nlvl:
-                # in the case of multi-fidelity optimization
-                # it is very probable that lower-fidelity correlation matrix
-                # becomes ill-conditionned
-                nugget = 10.0 * nugget
-        elif self.name in ["MGP"]:
-            nugget = 100.0 * nugget
+        nugget = self.options["nugget"]
+        if self.options["eval_noise"]:
+            nugget = 0
+        else:
+            if not self.options["nugget"]:
+                MACHINE_EPSILON = np.finfo(np.double).eps
+                nugget = 1e2 * MACHINE_EPSILON
+
         noise = 0
         tmp_var = theta
-        if self.options["eval_noise"]:
-            theta = tmp_var[:-1]
-            noise = tmp_var[-1]
+        if self.options["is_noise_het"]:
+            noise = self.noise
+        elif self.options["eval_noise"] and not self.options["is_noise_het"]:
+            theta = tmp_var[0 : self.D.shape[1]]
+            noise = tmp_var[self.D.shape[1] :]
         r = self._correlation_types[self.options["corr"]](theta, self.D).reshape(-1, 1)
 
-        R = np.eye(self.nt) * (1.0 + nugget + noise)
+        if self.options["is_noise_het"]:
+            R = np.eye(self.nt) * (1.0 + nugget + noise / self.nt_reps)
+        else:
+            R = np.eye(self.nt) * (1.0 + nugget + noise)
```

Comment:
As before, homoscedastic is just heteroscedastic with only one value.
Please avoid if else structure